libyul/AsmAnalysis.cpp

```diff
@@ -103,18 +105,21 @@ vector<YulString> AsmAnalyzer::operator()(Literal const& _literal)
 		m_errorReporter.typeError(
 			3069_error,
 			_literal.location,
-			"String literal too long (" + to_string(_literal.value.str().size()) + " > 32)"
+			"String literal too long ({} > 32).",
+			_literal.value.str().size()
 		);
 	else if (_literal.kind == LiteralKind::Number && bigint(_literal.value.str()) > u256(-1))
-		m_errorReporter.typeError(6708_error, _literal.location, "Number literal too large (> 256 bits)");
+		m_errorReporter.typeError(6708_error, _literal.location, "Number literal too large (> 256 bits).");
 	else if (_literal.kind == LiteralKind::Boolean)
 		yulAssert(_literal.value == "true"_yulstring || _literal.value == "false"_yulstring, "");
 
 	if (!m_dialect.validTypeForLiteral(_literal.kind, _literal.value, _literal.type))
 		m_errorReporter.typeError(
 			5170_error,
 			_literal.location,
-			"Invalid type \"" + _literal.type.str() + "\" for literal \"" + _literal.value.str() + "\"."
+			"Invalid type {:q} for literal {:q}.",
```

Comment:
What is `:q`? String quoting? Does it only just enclose it in `"` does some escaping too?

Reply:
We did had helpers for `quoted(string)`. Not sure it is a good idea hiding this within YulString, when it is also heavily used in the TypeChecker. I would say perhaps a `quoted()` helper is actually more readable.

Reply:
Yeah, maybe a helper is better. Definitely less magical.

Reply:
I would actually say that we can get along without fmt because it essentially uses magic all the way.

Reply:
`printf`-level of magic is not that bad though. It's when you start defining custom stuff that it might get problematic.